reference/language-concepts/modules.markdown

```diff
@@ -0,0 +1,87 @@
+---
+layout: default
+title: Modules
+published: true
+tags: [language, concepts, syntax, modules]
+---
+
+Modules allow users to extend the capabilities of CFEngine in a modular way, they can be easily added and upgraded independently of when you upgrade your CFEngine version. Several different types of modules are available.
+
+{% comment %}
+## cfbs Modules
+
+cfbs (CFEngine Build System) Modules provide a way to share and consume CFEngine related assets. Policy, Modules and Data can all be distributed as cfbs modules.
+
+### Specification
+{% endcomment %}
+
+## Promise Modules
+
+Promise modules allow for the implementation of [*custom* promise types][promise-type-custom], extending the CFEngine Language. They communicate with `cf-agent` using the [*Promise Module Protocol*][promise-type-custom#protocol].
+
+**History:**
+
+* Introduced 3.18.0
```

Comment:
Introduced in 3.17, significant fixes and improvements in 3.18.

Reply:
thx, will fix introduced version. if you want to specify what was significant in 3.18 we can add that too.